addons/mail/models/mail_activity.py

```diff
@@ -399,14 +403,21 @@ def _search(self, domain, offset=0, limit=None, order=None, access_rights_uid=No
             f'"{self._table}"."id"',
             f'"{self._table}"."res_model"',
             f'"{self._table}"."res_id"',
+            f'"{self._table}"."user_id"',
         )
         self.env.cr.execute(query_str, params)
         rows = self.env.cr.fetchall()
 
         # group res_ids by model, and determine accessible records
+        # Note: the user can read all activities assigned to him
+        uid = self.env.uid
+        allowed_activity_ids = set()
         model_ids = defaultdict(set)
-        for _id, res_model, res_id in rows:
-            model_ids[res_model].add(res_id)
+        for activity_id, res_model, res_id, user_id in rows:
```

Comment:
If we change 'mail_activity_rule_user' to include own activities in read rule, normally we don't have to do changes here, do we ?